### PR TITLE
Update NFR and glossary

### PR DESCRIPTION
**Pull request by Imranr2**
Tackles part of https://github.com/AY2122S1-CS2103T-W16-2/tp/issues/206

### CODE REVIEW

**Review by zognin — COMMENTED** [review]
Minor nits

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -985,21 +985,27 @@ Priorities: High - (must have), Medium - (nice to have), Low -  (unlikely to hav
 1. Usability Requirements:
     * Should work on systems with Java 11 and above.
     * Should work on Windows, Linux and macOS.
+    * Should function fully offline without access to the internet.
+    * Cannot be used on mobile phones.
```
Comment (EuniceSim142) [inline docs/DeveloperGuide.md:989]: Can change this to devices to make it applicable for other devices like ipad etc.
```suggestion
    * Cannot be used on mobile devices.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1020,7 +1026,9 @@ Priorities: High - (must have), Medium - (nice to have), Low -  (unlikely to hav
 | of/    | ORDER_FINISHED           | Order           |
 | op/    | ORDER_PRICE              | Order           |
 | oq/    | ORDER_QUANTITY           | Order           |
+| r/     | RECIPE_INDEX             | Recipe          |
 | ri/    | RECIPE_INGREDIENT        | Recipe          |
+| rp/    | RECIPE_PRICE             | Recipe          |
 | rn/    | RECIPE_NAME              | Recipe          |
```
Comment (EuniceSim142) [inline docs/DeveloperGuide.md:1032]: Swap these 2, maintain lexicographical ordering
```suggestion
| rn/    | RECIPE_NAME              | Recipe          |
| rp/    | RECIPE_PRICE             | Recipe          |
```

**Review by EuniceSim142 — COMMENTED** [review]
LGTM, just a few nits.

**Review by zognin — APPROVED** [review]
LGTM thanks

**Pull request merged**